gulpfile.js

```diff
@@ -10,76 +10,134 @@
 
 'use strict';
 
-const depcheck = require('depcheck');
-const eslint = require('gulp-eslint');
+const depcheck_lib = require('depcheck');
+const eslint_lib = require('gulp-eslint');
 const fs = require('fs-extra');
 const gulp = require('gulp');
 const mergeStream = require('merge-stream');
 const mocha = require('gulp-mocha');
 const path = require('path');
 const runSeq = require('run-sequence');
-const tslint = require("gulp-tslint");
+const tslint_lib = require("gulp-tslint");
 const typescript = require('gulp-typescript');
 const typings = require('gulp-typings');
 
-const tsProject = typescript.createProject('tsconfig.json');
-
-gulp.task('init', () => gulp.src("./typings.json").pipe(typings()));
-
-gulp.task('lint', ['tslint', 'eslint', 'depcheck']);
-
-gulp.task('build', () =>
-  mergeStream(
-    gulp.src('src/**/*.ts').pipe(typescript(tsProject)),
-    gulp.src(['src/**/*', '!src/**/*.ts'])
-  ).pipe(gulp.dest('lib'))
-);
-
-gulp.task('clean', (done) => {
-  fs.remove(path.join(__dirname, 'lib'), done);
-});
-
-gulp.task('build-all', (done) => {
-  runSeq('clean', 'init', 'lint', 'build', done);
-});
-
-gulp.task('test', ['build'], () =>
-  gulp.src('test/**/*_test.js', {read: false})
-      .pipe(mocha({
-        ui: 'tdd',
-        reporter: 'spec',
-      }))
-);
-
-gulp.task('tslint', () =>
-  gulp.src('src/**/*.ts')
-    .pipe(tslint({
-      configuration: 'tslint.json',
-    }))
-    .pipe(tslint.report('verbose')));
-
-gulp.task('eslint', () =>
-  gulp.src('test/**/*.js')
-    .pipe(eslint())
-    .pipe(eslint.format())
-    .pipe(eslint.failAfterError()));
-
-gulp.task('depcheck', () => new Promise((resolve, reject) => {
-  depcheck(__dirname, {ignoreDirs: []}, (result) => {
-    let invalidFiles = Object.keys(result.invalidFiles) || [];
-    let invalidJsFiles = invalidFiles.filter((f) => f.endsWith('.js'));
-    if (invalidJsFiles.length > 0) {
-      console.log('Invalid files:', result.invalidFiles);
-      reject(new Error('Invalid files'));
-      return;
-    }
+function task(name, deps, impl) {
+  if (gulp.hasTask(name)) {
+    throw new Error(
+        `A task with the name ${JSON.stringify(name)} already exists!`);
+  }
+  gulp.task(name, deps, impl);
+}
+
+module.exports.init = function() {
+  task('init', () => gulp.src("./typings.json").pipe(typings()));
```

Comment:
I think there's some discussion to be had about how to reuse tasks.

You're going for:
```javascript
toolsTasks.init();
```

But just exposing the functions might be more clear from a gulpfile:
```javascript
gulp.task('init', toolsTasks.init);
gulp.task('depcheck', toolsTasks.depcheck({}));
```

And there are more complex options, maybe something like:
```javascript
toolsTasks = toolsCommon.tasks({prefix: 'polymer'});
toolsTasks.task('init'); // calls gulp.task('polymer:init');
toolsTasks.task(['tslint', 'eslint']);
toolsTask.task(toolsCommon.depcheck({}); // calls gulp.task('polymer:depcheck');
toolsTask.task('build', toolsCommon.build({}); // calls gulp.task('build');
```

I only bring that up because I'm in the middle of building a library of tasks for external users. For internal use I think the explicit case (my second example) might be best...

WDYT?